src/components/calcite-date/calcite-date-picker.tsx

```diff
@@ -68,14 +78,17 @@ export class CalciteDatePicker {
   /**
    * Active date.
    */
-  @State() activeDate = isNaN(Date.parse(this.value)) ? new Date() : new Date(this.value);
+  @State() activeDate = isNaN(Date.parse(this.value)) ? new Date() :
+  new Date(new Date(this.value).getUTCFullYear(), new Date(this.value).getUTCMonth(), new Date(this.value).getUTCDate());
 
   @Watch('value') 
-  onNameChanged(newValue: string) {
+  onNameChanged(newValue: string) {    
     if(!isNaN(Date.parse(newValue))){
-      this.activeDate = new Date(newValue);
+      this.valueAsDate = new Date(new Date(newValue).getUTCFullYear(), new Date(newValue).getUTCMonth(), new Date(newValue).getUTCDate());  
+      this.activeDate = new Date(new Date(newValue).getUTCFullYear(), new Date(newValue).getUTCMonth(), new Date(newValue).getUTCDate());
```

Comment:
since you're doing this above as well (get the year, month, date) you could make a private method that generates this for you like:

```
this.valueAsDate = this.generateDate(newValue);
this.activeDate = this.generateDate(newValue);

private generateDate(value): Date {
  const date = new Date(value);
  return new Date(date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDate());  
}
```

Reply:
Added the comments @paulcpederson